tests/test_decorators.py

```diff
@@ -518,3 +518,33 @@ def validate_many(self, data, many):
         errors = schema.validate([{'foo': 3, 'bar': 'not an int'}], many=True)
         assert 'bar' in errors[0]
         assert '_schema' not in errors
+
+    def test_allow_reporting_field_errors_in_schema_validator(self):
+
+        class NestedSchema(Schema):
+            baz = fields.Int(required=True)
+
+        class MySchema(Schema):
+            foo = fields.Int(required=True)
+            bar = fields.Nested(NestedSchema, required=True)
+            bam = fields.Int(required=True)
+
+            @validates_schema(skip_on_field_errors=True)
+            def consistency_validation(self, data):
+                errors = {}
+                if data['bar']['baz'] != data['foo']:
+                    errors['bar'] = {'baz': 'Non-matching value'}
+
+                if data['bam'] > data['foo']:
+                    errors['bam'] = 'Value should be less than foo'
+
+                if errors:
+                    raise ValidationError(errors)
+
+        schema = MySchema()
+        errors = schema.validate({'foo': 2, 'bar': {'baz': 5}, 'bam': 6})
```

Comment:
I think I see the issue. [You pointed out](https://github.com/marshmallow-code/marshmallow/issues/441#issuecomment-230115307) that the `messages` attribute says:

> if [it is] a dict, the keys will be field names and the values will be lists of messages.

This documentation is misleading, because it is seems to actually be referring to the way [`Unmarshaller.run_validator()` internally builds `errors`](https://github.com/marshmallow-code/marshmallow/blob/dev/marshmallow/marshalling.py#L198) by copying the message for each field name, not how you have to build your `messages` dictionary or how it is going to transform your `messages` dictionary. I think this comment probably should have been added to `Schema.load()` considering the implementation.

The implementation of [`Unmarshaller.run_validator()`](https://github.com/marshmallow-code/marshmallow/blob/dev/marshmallow/marshalling.py#L198) explicitly allows an arbitrary `message` dictionary argument to pass through as an "error message" value.

> Point me into documentation where it shows raising dicts in ValidationError and expecting it to not be treated as nested field errors.

Is it documented that passing a dictionary to the `message` argument of `ValidationError` is supposed to treat the keys as field names? I don't think that the `messages` attribute of `ValidationError` was actually meant to suggest that behavior.